snap/implicit.go

```diff
@@ -79,3 +85,54 @@ func makeImplicitSlot(snapInfo *Info, ifaceName string) *SlotInfo {
 		Interface: ifaceName,
 	}
 }
+
+// AddImplicitHooks adds hooks from the installed snap's hookdir to the snap info.
+//
+// Existing hooks (i.e. ones defined in the YAML) are not changed; only missing
+// hooks are added.
+func AddImplicitHooks(snapInfo *Info) error {
```

Comment:
Should this be part of the public API?  It sounds like whatever reads out a *snap.Info should return it complete.